cas-server-core-services/src/test/java/org/apereo/cas/services/AbstractResourceBasedServiceRegistryDaoTests.java

```diff
@@ -0,0 +1,391 @@
+package org.apereo.cas.services;
+
+import com.google.common.collect.Sets;
+import org.apache.commons.io.FileUtils;
+import org.apereo.cas.authentication.principal.ShibbolethCompatiblePersistentIdGenerator;
+import org.apereo.cas.services.support.RegisteredServiceRegexAttributeFilter;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.springframework.core.io.ClassPathResource;
+
+import java.net.URI;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.*;
+
+/**
+ * This is {@link AbstractResourceBasedServiceRegistryDaoTests}.
+ *
+ * @author Misagh Moayyed
+ * @since 5.0.0
+ */
+public abstract class AbstractResourceBasedServiceRegistryDaoTests {
+    public static final ClassPathResource RESOURCE = new ClassPathResource("services");
+
+    protected ServiceRegistryDao dao;
+
+    @BeforeClass
+    public static void prepTests() throws Exception {
+        FileUtils.cleanDirectory(RESOURCE.getFile());
+    }
+
+    @Test
+    public void checkLoadingOfServiceFiles() throws Exception {
+        prepTests();
+        verifySaveAttributeReleasePolicyMappingRules();
+        verifySaveAttributeReleasePolicyAllowedAttrRulesAndFilter();
+        assertEquals(this.dao.load().size(), 2);
+    }
+    
+    @Test
+    public void checkSaveMethodWithNonExistentServiceAndNoAttributes() {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setName("testSaveMethodWithNonExistentServiceAndNoAttributes");
+        r.setServiceId("testId");
+        r.setTheme("theme");
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The String literal "theme" appears 7 times in this file; the first occurrence is on line 54](https://www.codacy.com/app/mmoayyed/cas/file/2643672654/issues/source?bid=3256115&fileBranchId=3322154#l54)